### ISSUE TITLE
Additional Custom Rules

### ISSUE THREAD

**Issue by JB-Scott** [rich-newman/typescript-analyzer#24]
Thanks for the great extension. The base rules are a good start, but we like to use the additional rules provided by tslint-consistent-codestyle. I have been trying to get this to run, but the fact that you unzip everything to a temporary directory has been creating problems. If we place the 3 modules (tslib, tsutils, and tslint-consistent-codestyle) in the node_modules folder within your temp folder, and add "extends": [ "tslint-consistent-codestyle" ], to the tslint.json file, it works perfectly. However, it being a temp folder, it gets overwritten and we lose the modules we added. We also tried a "rulesDirectory" entry in tslint.json instead, with the modules located in a more permanent folder, but then the references to the tslib and tsutils modules from within tslint-consistent-codestyle no longer resolve. Do you have any suggestions on how we might be able to get this working without having to add the modules back to the temp folder every time it is overwritten? Thanks.

**Comment by rich-newman** [comment]
You're right, we've got JavaScript dependency issues.  There is a workaround.  Your rules directory approach works if you install locally the exact same versions of typescript and tslint that the Analyzer is using.  So currently do:

`npm install typescript@2.8.1 tslint@5.9.1 tslint-consistent-codestyle`

Then, for example, a tslint.json as below will have the prefer-while rule working

`{
  "rulesDirectory": [ "tslint-consistent-codestyle" ],
  "rules": {
    "prefer-while":  true
  }
}`


I admit that's an unsatisfactory workaround as it forces you to use a specific TypeScript version, and one that is likely to change when the Analyzer is released.  I'm not averse to adding the tslint-consistent-codestyle package to the TypeScript Analyzer release if there's some demand, as I'm thinking of doing the same for codelyzer, which has similar issues.

**Comment by JB-Scott** [comment]
Thank you for the response. Through some trial an error a few months ago, we did end up setting things up as you describe above. It is working for us at the moment, but as you said, a bit harder to maintain when changes are made. If others are interested as well, we would certainly like to see tslint-consistent-codestyle added to your package - it has some nice checks that we find very helpful.
